integration-test/src/main/java/com/sequenceiq/it/cloudbreak/actor/CloudbreakUserCache.java

```diff
@@ -1,66 +1,152 @@
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
 import com.sequenceiq.cloudbreak.common.json.JsonUtil;
 import com.sequenceiq.cloudbreak.util.FileReaderUtils;
+import com.sequenceiq.it.cloudbreak.cloud.v4.CommonCloudProperties;
 import com.sequenceiq.it.cloudbreak.exception.TestFailException;
 
+@Component
 public class CloudbreakUserCache {
 
-    private static volatile CloudbreakUserCache instance;
-
-    private static Object mutex = new Object();
+    private static final Logger LOGGER = LoggerFactory.getLogger(CloudbreakUserCache.class);
 
     private Map<String, List<CloudbreakUser>> usersByAccount;
 
-    private CloudbreakUserCache() {
+    private Map<String, String> initializedKeys;
+
+    private Set<String> accountIds;
+
+    @Value("${integrationtest.user.realUms.accountKey:default}")
+    private String realUmsUserAccount;
+
+    @Value("${integrationtest.user.realUms.environmentKey:dev}")
+    private String realUmsUserEnvironment;
+
+    @Inject
+    private CommonCloudProperties commonCloudProperties;
+
+    protected CommonCloudProperties commonCloudProperties() {
+        return commonCloudProperties;
     }
 
-    public static CloudbreakUserCache getInstance() {
-        CloudbreakUserCache result = instance;
-        if (result == null) {
-            synchronized (mutex) {
-                result = instance;
-                if (result == null) {
-                    result = new CloudbreakUserCache();
-                    instance = result;
-                }
-            }
-        }
-        return result;
+    public CloudbreakUser getByDisplayName(String name) {
+        return getByDisplayName(name, commonCloudProperties.getUser().getRealUms().getEnvironmentKey(),
+                commonCloudProperties.getUser().getRealUms().getAccountKey());
     }
 
-    public CloudbreakUser getByName(String name) {
-        if (usersByAccount == null) {
-            initUsers();
+    public CloudbreakUser getByDisplayName(String name, String environmentKey, String accountKey) {
+        setRealUmsUserEnvironment(environmentKey);
+        setRealUmsUserAccount(accountKey);
+
+        if (usersByAccount == null || !initializedKeys.containsKey(environmentKey) || !initializedKeys.containsValue(accountKey)) {
+            initUsers(environmentKey, accountKey);
         }
-        return usersByAccount.values().stream().flatMap(Collection::stream)
-                .filter(u -> u.getDisplayName().equals(name)).findFirst().get();
+        CloudbreakUser user = usersByAccount.values().stream().flatMap(Collection::stream)
+                .filter(u -> u.getDisplayName().equals(name)).findFirst()
+                .orElseThrow(() -> new TestFailException(String.format("There is no real UMS user with name: %s in account: %s", name, accountKey)));
+        LOGGER.info(" Real UMS user has been found in repository:: \nDisplay name: {} \nCrn: {} \nAccess key: {} \nSecret key: {} \nAdmin: {} ",
```

Comment:
This "repository" is misleading for me, somewhat implies there is a JPA repository in the picture.

Reply:
No worries, I can change that part of the message if it is strange for you.